### DISCUSSION TITLE
How can this library be made to support additional schema formats?

### DISCUSSION THREAD

**Discussion by mike-schenk** [LEGO/AsyncAPI.NET#134] [Q&A] [answered]
I have an AsyncAPI document that this library doesn't read properly because it uses `schemaFormat: application/vnd.oai.openapi;version=3.0.0`, which is one of the values that the AsyncAPI spec says implementations are recommended, but not required, to support.

The reader's diagnostics include the message
> `'application/vnd.oai.openapi;version=3.0.0' is not a supported format. Supported formats are application/vnd.aai.asyncapi+json, application/vnd.aai.asyncapi+yaml, application/vnd.aai.asyncapi, application/schema+json;version=draft-07, application/schema+yaml;version=draft-07`

Is there an extensibility point in the Reader that will allow me to introduce support for other schema formats?

**Comment by VisualBean** [comment]
Sorry i didn't catch up with this earlier.

The current implementation is coded against the [asyncapi schema format](https://v2.asyncapi.com/docs/reference/specification/v2.6.0#schemaObject), Which differs from openapi v3 for some fields and validations.
Ill look a bit deeper into it however

**Comment by VisualBean** [comment, marked as answer]
Hi @mike-schenk 
Ive now refactored and added support for Avro Schemas.

OAS should be easy to implement at this point.
see #170 

Basically what you will need, is this;

1. Add the Schema classes (probably almost a direct copy of the `AsyncApiSchema`).
2. Add an AsyncApiOAIMessagePayload class that decorates the new schema type.
3. Add a deserializer (probably almost a direct copy of the `AsyncApiSchemaDeserializer`).
4. Add the supported schemaformat types to the `AsyncApiMessageDeserializer` and forward the payload loading to the prior mentioned deserializer.
5. Write tests.
